control-plane/api/v1alpha1/terminatinggatewayservice_webhook.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	capi "github.com/hashicorp/consul/api"
+	admissionv1 "k8s.io/api/admission/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:object:generate=false
+
+type TerminatingGatewayServiceWebhook struct {
+	client.Client
+	ConsulClient *capi.Client
+	Logger       logr.Logger
+	decoder      *admission.Decoder
+	//ConsulMeta   common.ConsulMeta
+}
+
+// NOTE: The path value in the below line is the path to the webhook.
+// If it is updated, run code-gen, update subcommand/controller/command.go
+// and the consul-helm value for the path to the webhook.
+//
+// NOTE: The below line cannot be combined with any other comment. If it is
+// it will break the code generation.
+//
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-v1alpha1-terminatinggatewayservices,mutating=true,failurePolicy=fail,groups=consul.hashicorp.com,resources=terminatinggatewayservices,versions=v1alpha1,name=mutate-terminatinggatewayservices.consul.hashicorp.com,sideEffects=None,admissionReviewVersions=v1beta1;v1
+
+func (v *TerminatingGatewayServiceWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Similarly, can we add a test for the webhook as well? You don't need all the validation cases covered by the _types.go file, instead you can just test the case mentioned here: "If any terminating gateway service resource has the same service name as this one, reject it."

Reply:
Yes. Done